### PR TITLE
Make Camera Monitor Great Again

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

I don’t know why it was necessary to make the cameras the way they are now, the need to connect to each server, not a user-friendly interface, and also a terrible artificial delay as if the transmission was taking place via 1st generation Bluetooth. In short, 

I did this thing a long time ago, but then I broke the game with it, then I killed it. 

Now, after the crew monitoring rework, I tried it again and it works much better.

## Why / Balance

This will allow people to start using cameras (previously only masochists and people with a mania for spying on the captain did this)

## Technical details

This is a terrible shitcode and copy-paste, I will understand if you close it.
shitcode

## Media

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: VigersRay
- tweak: Improved camera monitoring.

### CODE REVIEW

**Comment by chromiumboy** [comment]
I like the concept, but yeah, the execution needs some work. Namely, don't copy paste the crew monitor UI code. Like you have the coordinates listed for the camera but they aren't needed in this context... Making the nav map UI visually distinct from the other monitors would be nice. Like changing up the colors and markers for the cameras for example

Also why do we still need to manually refresh subnets and cameras, this functionality can just be automated

**Comment by Baptr0b0t** [comment]
The UI map was exactly what I needed for my AI PR, hope to see it merged soon.
https://github.com/space-wizards/docs/pull/162
https://github.com/space-wizards/space-station-14/pull/25267

**Comment by VigersRay** [comment]
> I like the concept, but yeah, the execution needs some work. Namely, don't copy paste the crew monitor UI code. Like you have the coordinates listed for the camera but they aren't needed in this context... Making the nav map UI visually distinct from the other monitors would be nice. Like changing up the colors and markers for the cameras for example
> 
> Also why do we still need to manually refresh subnets and cameras, this functionality can just be automated

After the PR, I thought about camera colors by department and removing subnet update buttons. In general, you can add automatic detection of all cameras and subnets every 10 seconds, I think this will not be too much of a burden. The only thing is that you first need to update the subnets, wait until the network emission works, and then update the cameras.

File: Content.Client/SurveillanceCamera/UI/SurveillanceCameraMonitorBoundUi.cs [hunk 1]
```diff
@@ -14,18 +14,30 @@ public sealed class SurveillanceCameraMonitorBoundUserInterface : BoundUserInter
...
+    private readonly IEntityManager _entManager;
 
     public SurveillanceCameraMonitorBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
     {
         _eyeLerpingSystem = EntMan.System<EyeLerpingSystem>();
         _surveillanceCameraMonitorSystem = EntMan.System<SurveillanceCameraMonitorSystem>();
+        IoCManager.InjectDependencies(this);
+        _entManager = IoCManager.Resolve<IEntityManager>();
+        _eyeLerpingSystem = _entManager.EntitySysManager.GetEntitySystem<EyeLerpingSystem>();
+        _surveillanceCameraMonitorSystem = _entManager.EntitySysManager.GetEntitySystem<SurveillanceCameraMonitorSystem>();
```
Comment (ficcialfaint) [inline Content.Client/SurveillanceCamera/UI/SurveillanceCameraMonitorBoundUi.cs:26]: you have `EntMan` and `EntMan.System<T>`

File: Content.Client/SurveillanceCamera/UI/SurveillanceCameraNavMapControl.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        foreach ((var netEntity, var blip) in TrackedEntities)
+        {
+            if (netEntity != Focus)
+                continue;
+
+            if (!LocalizedNames.TryGetValue(netEntity, out var name))
+                name = "Unknown";
+
```
Comment (ficcialfaint) [inline Content.Client/SurveillanceCamera/UI/SurveillanceCameraNavMapControl.cs:65]: localization

File: Content.Server/SurveillanceCamera/Systems/SurveillanceCameraMonitorSystem.cs [hunk 3]
```diff
@@ -87,14 +89,14 @@ private void OnComponentStartup(EntityUid uid, SurveillanceCameraMonitorComponen
...
-        }
-    }
+    // private void OnSubnetRequest(EntityUid uid, SurveillanceCameraMonitorComponent component,
+    //     SurveillanceCameraMonitorSubnetRequestMessage args)
+    // {
+    //     if (args.Session.AttachedEntity != null)
+    //     {
+    //         SetActiveSubnet(uid, args.Subnet, component);
+    //     }
+    // }
```
Comment (ficcialfaint) [inline Content.Server/SurveillanceCamera/Systems/SurveillanceCameraMonitorSystem.cs:99]: don't comment out, delete instead

File: Resources/Locale/en-US/surveillance-camera/surveillance-camera-ui.ftl [hunk 4]
```diff
@@ -7,6 +7,8 @@ surveillance-camera-monitor-ui-status-connecting = Connecting:
 surveillance-camera-monitor-ui-status-connected = Connected:
 surveillance-camera-monitor-ui-status-disconnected = Disconnected
 surveillance-camera-monitor-ui-no-subnets = No Subnets
+surveillance-camera-monitor-ui-flavor-left = Dont check captain bathroom
```
Comment (ficcialfaint) [inline Resources/Locale/en-US/surveillance-camera/surveillance-camera-ui.ftl:10]:
```suggestion
surveillance-camera-monitor-ui-flavor-left = Don't check captain bathroom
```

**Review by ficcialfaint — CHANGES_REQUESTED** [review]

**Comment by Strol20** [comment]
Is this pr active?

**Comment by metalgearsloth** [comment]
Re-open if you come back to it.

**Pull request closed without merging**